backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java

```diff
@@ -441,11 +445,9 @@ private String getSpecimenTypeSnomed(String input) {
     return null;
   }
 
-  private String getSpecimenTypeDescription(String input) {
-    if (input.matches(ALPHABET_REGEX)) {
-      return input;
-    } else if (input.matches(SNOMED_REGEX)) {
-      return resultsUploaderCachingService.getSNOMEDToSpecimenTypeNameMap().get(input);
+  private String getSpecimenTypeDescription(String loinc) {
+    if (loinc != null) {
+      return resultsUploaderCachingService.getSNOMEDToSpecimenTypeNameMap().get(loinc);
     }
     return null;
   }
```

Comment:
The change here doesn't alter the behavior, just better null handling and the removal of unnecessary regex matching